tools/git-change-exec/actions.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"os"
+	"os/exec"
+	"reflect"
+)
+
+type action interface {
+	match(path string) bool
+	do() error
+}
+
+func id(i any) string {
+	ty := reflect.TypeOf(i)
+	if ty.Name() == "" {
+		ty = reflect.TypeOf(i).Elem()
+	}
+	return ty.Name()
+}
+
+func execCmdWithDefaults(name string, args ...string) *exec.Cmd {
+	cmd := exec.Command(name, args...)
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Stdin = os.Stdin
+
+	return cmd
+}
+
+// Do not forget to add your Action HERE
+var actions = map[string][]action{
+	"test": []action{
+		pillarTestAction{},
+		gitChangeExecTest{},
+		getDepsTestAction{},
+		bpftraceCompilerExecTest{},
+	},
+	"lint": []action{
```

Comment:
I don't see any invocation of the tool with "lint" as the argument. Is that missing or hiding somewhere?

Reply:
So far I would keep it to be run by the user, but I can also add a Makefile target, perhaps `lint`.